src/net/unix/stream.rs

```diff
@@ -59,4 +85,71 @@ impl UnixStream {
     pub async fn write<T: IoBuf>(&self, buf: T) -> crate::BufResult<usize, T> {
         self.inner.write(buf).await
     }
+
+    /// Attempts to write an entire buffer to the stream.
+    ///
+    /// This method will continuously call [`write`] until there is no more data to be
+    /// written or an error is returned. This method will not return until the entire
+    /// buffer has been successfully written or an error has occurred.
+    ///
+    /// If the buffer contains no data, this will never call [`write`].
+    ///
+    /// # Errors
+    ///
+    /// This function will return the first error that [`write`] returns.
+    ///
+    /// [`write`]: Self::write
+    pub async fn write_all<T: IoBuf>(&self, mut buf: T) -> crate::BufResult<(), T> {
```

Comment:
This is fine for now, but I'd like to refactor these ops behind a trait at some point.

Reply:
If you suggest the trait name and the module path it comes from, I'll create a PR. I seem to recall a trait was used in the std or the tokio crates.